protocols/imap/src/main/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParser.java

```diff
@@ -0,0 +1,127 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.decode.parser;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSet.Builder;
+import org.apache.james.imap.api.ImapCommand;
+import org.apache.james.imap.api.ImapConstants;
+import org.apache.james.imap.api.ImapMessage;
+import org.apache.james.imap.api.display.HumanReadableText;
+import org.apache.james.imap.api.process.ImapSession;
+import org.apache.james.imap.decode.ImapRequestLineReader;
+import org.apache.james.imap.decode.base.AbstractImapCommandParser;
+import org.apache.james.imap.message.request.GetAnnotationRequest;
+import org.apache.james.imap.message.request.GetAnnotationRequest.Depth;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.protocols.imap.DecodingException;
+
+import java.util.Set;
+
+public class GetAnnotationCommandParser extends AbstractImapCommandParser {
+    private static final CharMatcher ENDOFLINE_PATTERN = CharMatcher.isNot('\n').and(CharMatcher.isNot('\r'));
+    private final static String MAXSIZE = "MAXSIZE";
+    private final static String DEPTH = "DEPTH";
+
+    public GetAnnotationCommandParser() {
+        super(ImapCommand.authenticatedStateCommand(ImapConstants.GETANNOTATION_COMMAND_NAME));
+    }
+
+    @Override
+    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader requestReader, String tag, ImapSession session)
+            throws DecodingException {
+        GetAnnotationRequest.GetAnnotationRequestBuilder builder = new GetAnnotationRequest.GetAnnotationRequestBuilder(tag, command);
+        try {
+            builder.mailboxName(requestReader.mailbox());
+
+            while (requestReader.nextNonSpaceChar() == '(') {
```

Comment:
Here you should have a first while that reads the MAXSIZE and DEPTH options.
Then try to read the keys if present.

```
while (requestReader.nextNonSpaceChar() == '(') {
   // TO BE extracted
   switch (requestReader.next()) {
      case M: 
            return builder.maxSize(...);
      case D:
            ....
   }
   // Other while to handle request reading...
}
```

Just a proposal...